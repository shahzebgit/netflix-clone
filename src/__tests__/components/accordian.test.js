import { render, fireEvent } from "@testing-library/react";
import { Accordion } from "../../components";
import faqsData from "../../fixtures/faqs.json";

describe("<Accordion/>", () => {
  it("render the <Accordion/> with populated data ", () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Frame>
          {faqsData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Frame>
      </Accordion>
    );
    expect(getByText("Frequently Asked Questions")).toBeTruthy();
    expect(getByText("What is Netflix?")).toBeTruthy();
    expect(getByText("How much does Netflix cost?")).toBeTruthy();
    expect(getByText("Where can I watch?")).toBeTruthy();
    expect(getByText("How do I cancel?")).toBeTruthy();
    expect(getByText("What can I watch on Netflix?")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
