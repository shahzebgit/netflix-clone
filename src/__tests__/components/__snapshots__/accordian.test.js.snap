// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion/> render the <Accordion/> with populated data  1`] = `
<div
  class="accordion-styles__Container-sc-5rhp7y-0 cUHHrA"
>
  <div
    class="accordion-styles__Inner-sc-5rhp7y-2 jPsaGY"
  >
    <h1
      class="accordion-styles__Title-sc-5rhp7y-3 lgYoDb"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion-styles__Frame-sc-5rhp7y-1 gzbYRV"
    >
      <div
        class="accordion-styles__Item-sc-5rhp7y-5 kRSIpi"
      >
        <div
          class="accordion-styles__Header-sc-5rhp7y-4 lbtfOD"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion-styles__Body-sc-5rhp7y-6 ibovdH closed"
        >
          <span>
            Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices.

You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!
          </span>
        </div>
      </div>
      <div
        class="accordion-styles__Item-sc-5rhp7y-5 kRSIpi"
      >
        <div
          class="accordion-styles__Header-sc-5rhp7y-4 lbtfOD"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion-styles__Body-sc-5rhp7y-6 ibovdH closed"
        >
          <span>
            Watch Netflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from £5.99 a month. No extra costs or contracts.
          </span>
        </div>
      </div>
      <div
        class="accordion-styles__Item-sc-5rhp7y-5 kRSIpi"
      >
        <div
          class="accordion-styles__Header-sc-5rhp7y-4 lbtfOD"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion-styles__Body-sc-5rhp7y-6 ibovdH closed"
        >
          <span>
            Watch anywhere, anytime, on an unlimited number of devices. Sign in with your Netflix account to watch instantly on the web at netflix.com from your personal computer or on any internet-connected device that offers the Netflix app, including smart TVs, smartphones, tablets, streaming media players and game consoles.

You can also download your favourite programmes with the iOS, Android, or Windows 10 app. Use downloads to watch while you're on the go and without an internet connection. Take Netflix with you anywhere.
          </span>
        </div>
      </div>
      <div
        class="accordion-styles__Item-sc-5rhp7y-5 kRSIpi"
      >
        <div
          class="accordion-styles__Header-sc-5rhp7y-4 lbtfOD"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion-styles__Body-sc-5rhp7y-6 ibovdH closed"
        >
          <span>
            Netflix is flexible. There are no annoying contracts and no commitments. You can easily cancel your account online in two clicks. There are no cancellation fees – start or stop your account at any time.
          </span>
        </div>
      </div>
      <div
        class="accordion-styles__Item-sc-5rhp7y-5 kRSIpi"
      >
        <div
          class="accordion-styles__Header-sc-5rhp7y-4 lbtfOD"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion-styles__Body-sc-5rhp7y-6 ibovdH closed"
        >
          <span>
            Netflix has an extensive library of feature films, documentaries, TV programmes, anime, award-winning Netflix originals, and more. Watch as much as you want, any time you want.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
